src/tools/chdman.cpp

```diff
@@ -2512,6 +2642,52 @@ static void do_extract_cd(parameters_map &params)
 			// resize the buffer for the track
 			buffer.resize((TEMP_BUFFER_SIZE / output_frame_size) * output_frame_size);
 
+			// GDROM .cue/.bin embeds pregap in output file
+			if (mode == MODE_GDROM_CUEBIN && (trackinfo.pregap > 0) && (trackinfo.pgdatasize > 0))
+			{
+				if ((tracknum == toc.numtrks-1) && (gdrom_identify_pattern(&toc) >= GDROM_TYPE_III))
+				{
+					// special case, 75 frames actual-zeroes, 150 frames data-zeroes with PQ and sync
+					outputoffs += 75 * trackinfo.pgdatasize;
+					trackinfo.frames -= 225;
+					discoffs += 225;
+
+					uint8_t buffer[2352];
+					static const uint8_t syncbytes[12] = {0x00, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00};
+					uint32_t physframeofs = trackinfo.physframeofs;
+
+					// populate CRC lookup table for EDC
+					edc_crctable_init();
+
+					for (int frame=0; frame < 150; frame++)
+					{
+						uint32_t msf = lba_to_msf(physframeofs);
+						memcpy(&buffer[0], syncbytes, 12);
+						buffer[12] = msf>>16;
+						buffer[13] = msf>>8;
+						buffer[14] = msf&0xff;
+						buffer[15] = 1; // mode_1
+						memset(&buffer[16], 0, 2048);
+						edc_generate(&buffer[0]);
+						ecc_generate(&buffer[0]);
```

Comment:
Why the two calls in a row?  Isn’t it supposed to checksum `EDC_MODE1_OFFSET` bytes starting at offest `0` and store them at `EDC_MODE1_OFFSET`?

Reply:
You first need to calculate the E**D**C because you need that data to correctly calculate the E**C**C.

Reply:
Both are required. They do different things. The order is important.